src/clusterfuzz/_internal/tests/core/bot/tasks/trials_test.py

```diff
@@ -206,3 +206,17 @@ def test_corrupted_config_file_is_ignored(self):
     """Ensure that a trial probability will not be overriden using a corrupted config file."""
     config_file_content = '[{"app_args": "--a1", "app_name": "app_1", "probability": 0.8]'
     self.source_side_test(config_file_content, 0.7, 'app_1', '-x', None)
+
+  def test_trial_ignored_if_there_is_contradiction(self):
+    """Ensure that a flag is not added if it contradicts an already added flag."""
+    config_file_content = [{
+        "app_args": "--c4",
+        "app_name": "app_4",
+        "probability": 1.0,
+        "contradicts": ["--c5"]
```

Comment:
See my comment above about symmetrical contradictions. This test case is an example for an asymmetric contradiction. If you turn it around, it won't get detected I think. Maybe add another test where they are turned around and enhance the logic as described in my other comment?

Reply:
done, added another test case